onnxscript/function_libs/torch_lib/ops/nn.py

```diff
@@ -347,6 +350,14 @@ def aten_celu(self: FLOAT, alpha: float = 1.0) -> FLOAT:
     return op.Celu(self, alpha=alpha)  # op.Celu only support float32
 
 
+@torch_op("aten::celu")
+def aten_celu_upcasted(self: TFloatUnlessFloat32, alpha: float = 1.0) -> TFloatUnlessFloat32:
```

Comment:
nit: hmm we are also downcasting from double. I know I started with the term "upcast" in my original issue, but maybe we should name it something else to callout the onnx dtype limitation? `aten_celu_dtype_hack`? Just an idea and we might need to come up with something more formal if there are more cases.

Reply:
Updating to type_promoted. wdyt?